interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1047 @@
+            else:
+                note_raw = ""
+
+            # attempt to split note and comment
+            if len(note_raw.split("\n", 1)) == 1:
+                comment[species.name] = ""
+                comment_str = ""
+                note[species.name] = note_raw
+            else:
+                comment[species.name] = "!\n"
+                note[species.name], comment_str = note_raw.split("\n", 1)
+
+            if len(f"{species.name} {note[species.name]}") > 24:
+                comment_str += "\n" + note[species.name]
+                note[species.name] = ""
+
+            comment_str = comment_str.replace("\n", "\n! ")
+            comment[species.name] = f"{comment[species.name]}! {comment_str}"
+
+        return note, comment
+
+    def build_nasa7_text(solution_species):
+        # first line has species name, space for notes/date, elemental composition,
+        # phase, thermodynamic range temperatures (low, high, middle), and a "1"
+        # total length should be 80
+        #
+        # Ex:
+        # C6 linear biradi  T04/09C  6.   0.   0.   0.G   200.000  6000.000 1000.        1
+        #  1.06841281E+01 5.62944075E-03-2.13152905E-06 3.56133777E-10-2.18273469E-14    2
+        #  1.43741693E+05-2.87959136E+01 3.06949687E+00 3.71386246E-02-5.95698852E-05    3
+        #  5.15924485E-08-1.77143386E-11 1.45477274E+05 8.35844575E+00 1.47610437E+05    4
+
+        # get notes and comments for all species
+        note, comment = build_thermo_notes(solution_species)
+
+        # write data for each species in the Solution object
+        thermo_text = []
+        for species in solution_species:
+            composition_string = "".join(
+                [f"{s:2}{int(v):>3}" for s, v in species.composition.items()]
+            )
+
+            # TODO: Currently hardcoded for gas phase
+            name_and_note = f"{species.name} {note[species.name]}"
+            text = [comment[species.name]]
+            text.append(
+                f"{name_and_note:<24}"  # name and date/note field
+                + f"{composition_string:<20}"
+                + "G"  # only supports gas phase
+                + f"{species.thermo.min_temp:10.3f}"
+                + f"{species.thermo.max_temp:10.3f}"
+                + f"{species.thermo.coeffs[0]:10.3f}"
+                + f"{1:>5}"
+            )
+
+            # second line has first five coefficients of high-temperature range,
+            # ending with a "2" in column 79
+            text.append(
+                "".join([f"{c:15.8e}" for c in species.thermo.coeffs[1:6]]) + f"{2:>5}"
+            )
+
+            # third line has the last two coefficients of the high-temperature range,
+            # first three coefficients of low-temperature range, and "3"
+            text.append(
+                "".join([f"{c:15.8e}" for c in species.thermo.coeffs[6:8]])
+                + "".join([f"{c:15.8e}" for c in species.thermo.coeffs[8:11]])
```

Comment:
I don't think you should do this in two lines.

Reply:
As in make it one long line or break it up even more?

Reply:
Sorry, that wasn't clear 😄 I think you can just do `species.thermo.coeffs[6:11]`.

Reply:
Hmm. Yes this is redundant. I think it must have been this way because coeffs were broken into high and low. Regardless, it's changed to what you suggested.